subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ExecuteActionsTaskExecuter.java

```diff
@@ -75,7 +79,10 @@ public void execute(TaskInternal task, TaskStateInternal state, TaskExecutionCon
                     state.getDidWork() ? TaskExecutionOutcome.EXECUTED : TaskExecutionOutcome.UP_TO_DATE
                 );
             }
-            context.getTaskArtifactState().snapshotAfterTaskExecution(failure);
+            context.getTaskArtifactState().snapshotAfterTaskExecution(failure, new OriginTaskExecutionMetadata(
+                buildInvocationScopeId.getId(),
+                context.markExecutionTime()
```

Comment:
The time recorded here does not include snapshotting the outputs. We need to keep this in mind if we calculate saved time.

Reply:
Changed so that it does now.